GitSearch/Networking/Encoding/URLEncoder.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  URLEncoder.swift
+//  GitSearch
+//
+//  Created by Vldmr on 8/26/19.
+//  Copyright © 2019 mudritskiy. All rights reserved.
+//
+
+import Foundation
+
+public struct URLEncoder {
+    
+    static func encodeParameters(for urlRequest: inout URLRequest, with parameters: HTTPParameters?) throws {
+        guard let url = urlRequest.url else { throw HTTPNetworkError.missingURL }
+        guard var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false), let parameters = parameters else { return }
+        if parameters.isEmpty { return }
```

Comment:
How com `if` made it where `guard` belongs?)

Reply:
It's not just a matter of which word to use, `guard` have strong guarantee that if precondition does not fulfill, the rest of the code will never be reached. Compiler ensures that.